core/src/Kokkos_Core_fwd.hpp

```diff
@@ -149,45 +149,61 @@ struct Device;
 /// Kokkos::Cuda, Kokkos::Experimental::OpenMPTarget, Kokkos::OpenMP,
 /// Kokkos::Threads, Kokkos::Serial
 
+#if defined(__clang_analyzer__)
+#define KOKKOS_IMPL_DEFAULT_EXEC_SPACE_ANNOTATION \
+  [[clang::annotate("DefaultExecutionSpace")]]
+#define KOKKOS_IMPL_DEFAULT_HOST_EXEC_SPACE_ANNOTATION \
+  [[clang::annotate("DefaultHostExecutionSpace")]]
```

Comment:
I assume `__attribute__((annotate("Foo"))` and `[[clang::annotate("Foo")]]` are equivalent.  Can you confirm it is valid syntax for all LLVM/Clang version that we support?

Reply:
I haven't checked which versions of clang support the `[[]]` syntax, but this won't prevent old clangs from compiling Kokkos, but it could impact clang-tidy and/or the static analyzer.  I'll see what I can do though to test.